HISTORY.md

```diff
@@ -1,3 +1,142 @@
+# TileDB v2.27.0 Release Notes
+
+## Disk Format
+
+* (documentation) The storage format specification was updated to document format changes of previous versions throughout the main document. [#5329](https://github.com/TileDB-Inc/TileDB/pull/5329)
+
+## Breaking behavior
+
+* Deleting an array or group will also delete its root directory if it was left empty. [#5269](https://github.com/TileDB-Inc/TileDB/pull/5269)
+
+## New features
+
+* Dumping fragment info now includes the name of each fragment's array schema. [#5257](https://github.com/TileDB-Inc/TileDB/pull/5257)
+* Added support for the `ssl.ca_path` config option on Azure. [#5286](https://github.com/TileDB-Inc/TileDB/pull/5286)
+* Connections to TileDB REST API now use TCP keepalive by default. [#5319](https://github.com/TileDB-Inc/TileDB/pull/5319) and curl retries [#5273](https://github.com/TileDB-Inc/TileDB/pull/5273).
+* Added support for "current domain" on dense arrays. [#5303](https://github.com/TileDB-Inc/TileDB/pull/5303)
+
+### Configuration Options
+
+* Add `vfs.log_operations` config option to log in trace level all VFS operations and their duration. [#5278](https://github.com/TileDB-Inc/TileDB/pull/5278)
+* The default value of `vfs.s3.region` was changed to empty, which will let the AWS pick the default region from sources such as environment variables, profile configuration, and instance metadata. [#5317](https://github.com/TileDB-Inc/TileDB/pull/5317)
+* Add `rest.curl.tcp_keepalive` config option that controls using TCP keepalive for TileDB REST connections. It is enabled by default. [#5319](https://github.com/TileDB-Inc/TileDB/pull/5319)
+
+## Improvements
+
+* Fix read queries on sparse arrays where only aggregates are requested and no layout is specified. [#5255](https://github.com/TileDB-Inc/TileDB/pull/5255)
+* Do not pass back RTrees to clients. [#5265](https://github.com/TileDB-Inc/TileDB/pull/5265)
+* Prevent potentially unsafe consolidation in dense arrays by fragment list. [#5251](https://github.com/TileDB-Inc/TileDB/pull/5251)
+* Improve error messages for unsupported data types in aggregates. [#5289](https://github.com/TileDB-Inc/TileDB/pull/5289)
+* Remove StorageManager from stats dumps. [#5297](https://github.com/TileDB-Inc/TileDB/pull/5297)
+* Precompute memory usage for tile offsets and error if not loadable due to memory limits. [#5310](https://github.com/TileDB-Inc/TileDB/pull/5310)
+* GCS object composition temporary files now end with `.tmp`, which may be used to set a lifecycle rule. [#5308](https://github.com/TileDB-Inc/TileDB/pull/5308)
+* Updated libmagic to version 5.45. [#5332](https://github.com/TileDB-Inc/TileDB/pull/5332)
+* Linux aarch64 release [#5271](https://github.com/TileDB-Inc/TileDB/pull/5271)
+* aarch64 release artifacts (manylinux2014 environment) [#5348](https://github.com/TileDB-Inc/TileDB/pull/5348)
+* Revert selective Rtree serialization to allow release testing [#5370](https://github.com/TileDB-Inc/TileDB/pull/5370)
```

Comment:
```suggestion
* Prevent potentially unsafe consolidation in dense arrays by fragment list. [#5251](https://github.com/TileDB-Inc/TileDB/pull/5251)
* Improve error messages for unsupported data types in aggregates. [#5289](https://github.com/TileDB-Inc/TileDB/pull/5289)
* Remove StorageManager from stats dumps. [#5297](https://github.com/TileDB-Inc/TileDB/pull/5297)
* Precompute memory usage for tile offsets and error if not loadable due to memory limits. [#5310](https://github.com/TileDB-Inc/TileDB/pull/5310)
* GCS object composition temporary files now end with `.tiledb.tmp`, which may be used to set a lifecycle rule. [#5372](https://github.com/TileDB-Inc/TileDB/pull/5372)
* Updated libmagic to version 5.45. [#5332](https://github.com/TileDB-Inc/TileDB/pull/5332)
* Linux aarch64 release [#5271](https://github.com/TileDB-Inc/TileDB/pull/5271)
```

* Removed the two RTree serialization changes (the first was reverted by the second and they cancel out).
* Updated temporary file extension and PR.
* Removed the manylinux2014 aarch64 change (it's a bug fix by a change introduced in the same release cycle).